### PR TITLE
ISPN-9366 Cache.size optimizations

### PR DESCRIPTION
**Pull request by jabolina**
* Created a dedicated class to handle the SizeCommand. This class mixes
some of the Publisher* methods.

* We query locally for the size and issue request to some of the remote
targets to do the same. With the returned value we sum everything.

* Change ComponentRegistry to create the LocalClusterPublisherManager,
only needed so the component is already started when used.

### CODE REVIEW

File: core/src/main/java/org/infinispan/commands/read/SizeCommand.java [hunk 1]
```diff
@@ -12,18 +28,108 @@
...
+      output.writeObject(segments);
+      MarshallUtil.marshallCollection(excludedKeys, output);
+      output.writeInt(topologyId);
+   }
+
+   @Override
+   public void readFrom(ObjectInput input) throws IOException, ClassNotFoundException {
+      setFlagsBitSet(input.readLong());
+      segments = (IntSet) input.readObject();
+      excludedKeys = MarshallUtil.unmarshallCollectionUnbounded(input, HashSet::new);
```
Comment (wburns) [inline core/src/main/java/org/infinispan/commands/read/SizeCommand.java:125]: This is missing the tpologyId. Needs to mirror write exactly.

File: core/src/main/java/org/infinispan/factories/SizeCommandHandlerFactory.java [hunk 2]
```diff
@@ -0,0 +1,12 @@
+package org.infinispan.factories;
+
+import org.infinispan.factories.annotations.DefaultFactoryFor;
+import org.infinispan.reactive.size.FallbackSizeCommandHandler;
+
+@DefaultFactoryFor(classes = FallbackSizeCommandHandler.class)
+public class SizeCommandHandlerFactory extends AbstractNamedCacheComponentFactory implements AutoInstantiableFactory {
+
+   @Override public Object construct(String name) {
+      return new FallbackSizeCommandHandler();
```
Comment (wburns) [inline core/src/main/java/org/infinispan/factories/SizeCommandHandlerFactory.java:10]: This seems unneded. Or was the point to add another impl based on the cache config?

File: core/src/main/java/org/infinispan/interceptors/impl/CallInterceptor.java [hunk 3]
```diff
@@ -604,16 +598,7 @@ public Object visitRemoveExpiredCommand(InvocationContext ctx, RemoveExpiredComm
...
-      if (command.hasAnyFlag(FlagBitSets.CACHE_MODE_LOCAL)) {
-         managerToUse = localClusterPublisherManager;
-      } else {
-         managerToUse = clusterPublisherManager;
-      }
-
-      return asyncValue(managerToUse.keyReduction(false, null, null, ctx,
-            command.getFlagsBitSet(), DeliveryGuarantee.EXACTLY_ONCE, PublisherReducers.count(),
-            PublisherReducers.add()));
+      return sizeHandler.size(ctx, command);
```
Comment (wburns) [inline core/src/main/java/org/infinispan/interceptors/impl/CallInterceptor.java:601]: I was thinking we would use the interceptors to instead handle if the cache was distributed for example. This way the `BaseDistributionInterceptor` would handle the SizeCommand in `visitSizeCommand` and the `CallInterceptor` would handle for local caches or when CACHE_MODE_LOCAL flag was set.

Comment (wburns) [reply core/src/main/java/org/infinispan/interceptors/impl/CallInterceptor.java:601]: The other option, which is more aligned with what you have here is instead to use the ClusterPublisherManager directly and add a method to it.  This way we wouldn't duplicate all the code in your `FallbackSizeCommandHandler` class.

File: core/src/main/java/org/infinispan/factories/SizeCommandHandlerFactory.java (hunk 2 above)
Comment (jabolina) [reply core/src/main/java/org/infinispan/factories/SizeCommandHandlerFactory.java]: ended up this was unnecessary

File: core/src/main/java/org/infinispan/interceptors/impl/CallInterceptor.java (hunk 3 above)
Comment (jabolina) [reply core/src/main/java/org/infinispan/interceptors/impl/CallInterceptor.java]: Went with the approach to create a method in the `ClusterPublisherManager`.

**Pull request closed without merging**